### ISSUE TITLE
How to handle "research"-like tasks?

### ISSUE THREAD

**Issue by skapral** [zerocracy/datum#262]
Working on 0crat, I noticed that it treats every member of the project equally, and can grant task of any ambiguity degree to any member of the project, and still with fixed cost. It leads sometimes to situations when assignee become a hostage of 0crat's random decision.

Consider the task https://github.com/yegor256/jpeek/issues/17 from JPeek. I haven't heard about this metric anything before. Issue doesn't provide much information on it. Project - either: that metric is not similar to anything which was done so far in JPeek: it is conceptual and based on lexical analysis of the sources. So, to start fixing the issue, I needed to study the metric from scratch, expecting help from noone. It took the most part of these 30 mins I will be rewarded for - and it was only studying and understanding the matters. Taking into account the time spent for that limited solution I provided - even just documenting all puzzlers and impediments - I spent way more than 30 minutes. I find this completely ineffective from PM point of view and discouraging from my point of view.

And looking back I see no other way of how could I have reduced the time spent there. Just glancing through the spec for 5 mins would not be enough: without understanding the matters I wouldn't be able to introduce sane puzzles and impediments. IMO same problem awaits for reviewer: if I hadn't spent time on understanding and somehow documenting my understanding related to the metric, reviewer would have no ability to reason objectively about sanity of my pull request and impediments without studying the spec. in detail for himself.

Honestly, I would gladly refuse from this "research" immediately, if not the paragraph [6](http://datum.zerocracy.com/pages/policy.html#6) from the 0crat policy. -15 points is very significant punishment, especially for me as a newcomer, so I was literally forced to stay with it.

Speaking about possible solution: I have two ideas in mind:
- to correlate somehow the boost factor with ambiguity of the task. It could be done by ARChitector when he puts task in scope, for example. Boost factor would compensate expences spent on research, but there is a danger of overpayment if the task is accidentally assigned to highly-paid project member.
- to rank tasks by ambiguity (by means of ARC desicion when scoping the tasks), and taking into account this rank when assigning the tasks (highly ranked tasks goes to dudes with more points or dudes who are long enough in a project).

PS: Or maybe I treated it wrong? Where have I mistaken then?

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@skapral did you read this? http://www.yegor256.com/2015/01/15/how-to-cut-corners.html

**Comment by skapral** [comment]
@yegor256 Yes, I did, and meditated a lot on it before joining ZC. But I don't understand how it would help me in this particular situation. What is my mistake?

Seems like my case is described best by section "Demand Better Documentation and Wait". But it is not so easy to do without knowing the subject. I already tried in a very beginning of these 30 minutes to create impediment on this issue. And you declined [it](https://github.com/yegor256/jpeek/issues/110), which made me forced to study more on the subject. And you are about to decline [another one](https://github.com/yegor256/jpeek/issues/129) - my only hope is on puzzle in the PR now. Your motivation is clear for me, I have no objections on it - indeed these impediments are too shady to accept. But to make them more clear, I am forced to do more research and spend more then 30 minutes.

PS: I have some doubts on where to raise this discussion. I placed it here because IMO it is not right for ZC to blindly pick up random contributors for such epic story. If you think that everything is ok, I can separate out the philosophical aspect of XDSD to somewhere else - either an article comments or XDSD GitHub repository.

**Comment by yegor256** [comment]
@skapral your concerns are valid. I'm thinking what would be the right scenario for this particular task.
